### DISCUSSION TITLE
Disable FCM for iOS

### DISCUSSION THREAD

**Discussion by er1cAk** [invertase/react-native-firebase#5290] [Q&A] [answered]
I just wanna ask, if (and how) I am able to disable FCM for ios... In my case I want to use FCM for android and for iOS access directly to the APNS. But every time if aj call function .getToken() I will successfully register iOS too.

**Comment by mikehardy** [comment, marked as answer]
Not sure why you'd want platform-specific code, but you could always use patch-package to delete the RNFBMessaging.podspec file, at which point react-native auto-linking won't see the module, so cocoapods won't install+link it and it will not initialize. Could work, totally untested though

**Reply by er1cAk** [reply]
@mikehardy thank you for your answer! 

I need platform specific code because I facing with delay (4-6s) on message(data-only noti) delivery when app is on background or killed. Also I am not able to set priority to 10, only to 5...  I'm based on apple documentation that I should be able to set priority to 10.. for app purpose it's really important to had message delivered immediately in most cases (I know in some cases I still wouldn't be able to achieve that).

Or do you know why I had delay on delivery?

**Reply by mikehardy** [reply]
Strange, I'm not sure where there is a delay. However for guaranteed delivery on iOS while still having a little code control prior to display, you might like implementing a notification service extension like https://notifee.app/react-native/docs/ios/remote-notification-support or https://rnfirebase.io/messaging/ios-notification-images - the doc here is focused on adding images but you can do something other than images if you want, though I believe you have a very tight time-slice of CPU / energy to do it in

**Reply by er1cAk** [reply]
Yeah, really thank you for your advice. I will try it.. btw on Android it works perfectly in all of states of device. I will let you know results. :)

**Reply by er1cAk** [reply]
@mikehardy

I found something, if background message should arrive to the device I see these [logs](https://gist.github.com/er1cAk/8703879f91bcec1367e97b1b239dcd73) in console... Do you have suspicion what could caused it?

**Reply by mikehardy** [reply]
No idea what those mean, sorry
